### ISSUE TITLE
eventDateSingle missing time

### ISSUE THREAD

**Issue by MortenHofft** [gbif/pipelines#220]
`eventDateSingle` might be better be named `startDate` if that is what it is.
Secondly I wonder why the time has been removed. As far as I know there is no option for sorting records by dateTime (since the range field do not allow it). That means we cannot visualise tracking data well.

```
"eventDateSingle": "2014-03-24",
"year": 2014,
"month": 3,
"day": 24,
"eventDate": {
    "gte": "2014-03-24T17:02:25"
},
```

**Comment by muttcg** [comment]
AFAK. This field was introduced to support backward compatibility with API v1, cause we don't have in v1, we don't populate here here either

**Comment by MortenHofft** [comment]
Makes sense. API v1 supports search by date alone, but we could add datetime in ES for v2 usage - v1 could just use the date part.

**Comment by timrobertson100** [comment]
I'd recommend bringing in a naming convention now, as this will surely be something that we hit as we aim to improve things.

One idea could be to document the schema in a README and bring in the concept of deprecating fields. Documenting why they are deprecated in the schema would be helpful:

1. Rename the field to `dep_eventDateSingle`  
2. Add new fields of `eventDateTimeStart`

An alternative could be to introduce `v2_...` nomenclature but that will just get messy quickly

**Comment by muttcg** [comment]
We already have:
eventDate - a new one and ES date_range way of dates, supports range queries and contains gte (start) and lte (end) dates, fixed ES structure 
eventDateSingle - support API v1

The data inside is the same, I don't think we need one more, moreover, we can delete eventDateSingle, cause eventDate.gte == eventDateSingle all the time

**Comment by muttcg** [comment]
@MortenHofft Explained, we can't sort by date_range, I think we can extend eventDateSingle to date and time

**Comment by asturcon** [comment]
For the API that change is ok. It already shows the time but it's always 00:00:00 since it doesn't come from ES: http://api.gbif.org/v1/occurrence/search?facet=eventDate&limit=0

**Comment by muttcg** [comment]
Fixed in 2.2.19
I reindexed manually one dataset by now - https://www.gbif.org/dataset/83e20573-f7dd-4852-9159-21566e1e691e
